functions/slack/signing_test.go

```diff
@@ -0,0 +1,75 @@
+package slack
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestGoodSignature(t *testing.T) {
+
+	secret := "talesfromthecrypt"
+	ts := strconv.FormatInt(time.Now().Unix(), 10)
+	body := "somebody"
+	base := fmt.Sprintf("v0:%s:%s", ts, body)
+	correctSHA2Signature := fmt.Sprintf("v0=%s", hex.EncodeToString(getSignature([]byte(base), []byte("talesfromthecrypt"))))
+
+	goodReq := httptest.NewRequest("POST", "https://someurl.com", strings.NewReader(body))
+	goodReq.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	goodReq.Header.Add("X-Slack-Request-Timestamp", ts)
+	goodReq.Header.Add("X-Slack-Signature", correctSHA2Signature)
+
+	goodResult, err := verifyRequestSignature(goodReq, secret)
+	if err != nil {
+		t.Errorf("TestGoodSignature got an error: %s", err)
+	}
+	if !goodResult {
+		t.Errorf("TestGoodSignature failed. Got: %v | Expected: %v", goodResult, !goodResult)
+	}
+}
+
+func TestBadSignature(t *testing.T) {
```

Comment:
Consider turning these into a [table driven test](https://github.com/golang/go/wiki/TableDrivenTests), with one test function for valid cases and another (table driven) test for invalid cases.

See https://github.com/golang/go/wiki/TestComments#table-driven-tests-vs-multiple-test-functions, in particular the last paragraph which talks about combining multiple test functions with table driven tests.